### DISCUSSION TITLE
Recommendations for md-dialog

### DISCUSSION THREAD

**Discussion by EarzuChan** [material-components/material-web#5161] [Ideas]
I hope to add a attribute in the future to allow the user to click on the outer to dismiss or not.
Another thing I found is that my CSS reset (the * selector) will directly affect the top-level style of the component style. Why?😂

**Comment by asyncliz** [comment]
You can turn off default cancelation gestures like clicking on the scrim, pressing escape, or performing a mobile back gesture by calling `event.preventDefault()` on the `cancel` event.

```ts
html`
  <md-dialog 
    @cancel=${event => event.preventDefault()}
    @close=${this.handleClose}
  >...</md-dialog>
`;
```

This is useful for critical dialogs that require a user choice.
